### PR TITLE
Closes #268: Move group provider references outside group entities

### LINKED ISSUE #268: Move GroupProvider references outside of Group entity classes

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java [hunk 1]
```diff
@@ -143,20 +139,14 @@ public ExplicitGroup(ExplicitGroupProvider prv) {
     }
 
     public Set<ExplicitGroup> getContainedExplicitGroups() {
-        if (getGroupProvider() != null) {
-            for (ExplicitGroup g : containedExplicitGroups) {
-                g.setProvider(getGroupProvider());
-            }
-        }
+//        if (getGroupProvider() != null) {
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: Why do we add commented code?

Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: commented code should be deleted

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java [hunk 2]
```diff
@@ -392,18 +310,9 @@ public boolean isEditable() {
...
-    }
-
-    void setProvider(ExplicitGroupProvider c) {
-        provider = c;
-    }
-
     @Override
     public String getIdentifier() {
-        return Group.IDENTIFIER_PREFIX + provider.getGroupProviderAlias()
+        return Group.IDENTIFIER_PREFIX + GROUP_TYPE
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java:310]: Let's make it GROUP_TYPE.toLowerCase() like in other places?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java (hunk 1 above)
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: commented code should be deleted

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroupProvider.java [hunk 3]
```diff
@@ -99,33 +109,71 @@ public ExplicitGroup makeGroup() {
...
+    private boolean containsIndirectly(DataverseRequest req, ExplicitGroup explicitGroup) {
+        for (ExplicitGroup ceg : explicitGroup.getContainedExplicitGroups()) {
+            if (this.contains(req, ceg)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    @SuppressWarnings("unchecked")
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroupProvider.java:173]: Can you add a comment explaining why warning is suppressed?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AllUsersGroupProvider.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package edu.harvard.iq.dataverse.authorization.groups.impl.builtin;
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AllUsersGroupProvider.java:1]: Can we cover new class with tests?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AuthenticatedUsersProvider.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package edu.harvard.iq.dataverse.authorization.groups.impl.builtin;
```
Comment (dbojanek) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AuthenticatedUsersProvider.java:1]: Can we cover new class with tests?

**Review by dbojanek — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java (hunk 1 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: ok

Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: ok

Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AuthenticatedUsersProvider.java (hunk 5 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AuthenticatedUsersProvider.java:1]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AllUsersGroupProvider.java (hunk 4 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/builtin/AllUsersGroupProvider.java:1]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroupProvider.java (hunk 3 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroupProvider.java:173]: ok

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java (hunk 2 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/explicit/ExplicitGroup.java:310]: I've added toLoweCase only in AllUsers and AuthenticatedUsers `getAlias` method. I've removed toLowerCase for them.

**Review by dbojanek — APPROVED** [review]

**Pull request merged**